oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -340,6 +343,23 @@ public AccessToken refreshAccessToken() throws IOException {
     HttpRequest request = requestFactory.buildPostRequest(new GenericUrl(tokenServerUri), content);
     request.setParser(new JsonObjectParser(jsonFactory));
 
+    request.setIOExceptionHandler(new HttpBackOffIOExceptionHandler(new ExponentialBackOff()));
+    request.setUnsuccessfulResponseHandler(
```

Comment:
The original [issue](https://github.com/google/google-auth-library-java/issues/91) is about IOException. I'm not sure how IOException and http errors are connected in this case (if connected at all), but please confirm that both (not just one) IO exception and http error response retries are necessary here and that they do not conflict with each other.